### PR TITLE
Automate building of packages in OBS

### PR DESCRIPTION
**Pull request by aazam-gh**
This PR is part of the Feilong Packaging Process to automate building of packages (zthin and zvmsdk) for the 3 distros (Ubuntu, RHEL, SLES) on OBS.

In order for this automation to work, we must use a webhook in order for OBS to receive events from the github action to trigger the rebuild of the packages with the latest source files. 
In order to use the webhook, we require a Personal Access Token (PAT) which will be used by OBS to generate the url for the webhook.

The working of the action is as follows: 
- When build-zvmsdk receives the dispatched event from this (feilong) github action, it first runs a few script files to generate new tarballs from the latest version of the feilong master branch.
- After creating all the source files, it downloads them as artifacts and pushes them to the master branch of build-zvmsdk.
- Once build-zvmsdk has a push, it triggers the webhook with OBS which causes each package hosted on OBS to download the required tarballs and/or .dsc files followed by triggering a rebuild for each package.

This [PR](https://github.com/openmainframeproject/feilong/pull/839) handles the development on `feilong` side.

Please create a PAT on github for this repository.
The webhook will then be created with the `Push` event.

### CODE REVIEW

**Review by jichenjc — APPROVED** [review]

File: .obs/workflows.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+workflow:
+  steps:
+    - trigger_services:
+        project: home:Aazam:feilong
```
Comment (Bischoff) [inline .obs/workflows.yml]: Don't forget to change this before merging :)

File: .obs/workflows.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        package: zthin_deb
+        package: zthin_rhel
+        package: zthin_sles
+        package: zvmsdk_deb
+        package: zvmsdk_rhel
+        package: zvmsdk_sles
+    - rebuild_package:
+        project: home:Aazam:feilong
```
Comment (Bischoff) [inline .obs/workflows.yml]: same

File: .github/buildzthin_deb_tar [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+# Copyright 2017 IBM Corp.
```
Comment (Bischoff) [inline .github/buildzthin_deb_tar]: For new files i'm not sure this copyright makes sense (?)

File: .github/buildzthin_tar [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+# Copyright 2017 IBM Corp.
```
Comment (Bischoff) [inline .github/buildzthin_tar]: same

File: .github/buildzvmsdk_deb_tar [hunk 5]
```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+# Copyright 2017 IBM Corp.
```
Comment (Bischoff) [inline .github/buildzvmsdk_deb_tar]: same

File: .github/buildzvmsdk_tar [hunk 6]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+# Copyright 2017 IBM Corp.
```
Comment (Bischoff) [inline .github/buildzvmsdk_tar]: same

**Review by Bischoff — COMMENTED** [review]
See the usual nitpicks

**Review by Bischoff — APPROVED** [review]
lgtm

**Review by bjhuangr — APPROVED** [review]

**Comment by aazam-gh** [comment]
@bjhuangr are you able to create a PAT on GitHub?

**Comment by bjhuangr** [comment]
> @bjhuangr are you able to create a PAT on GitHub?

I'm not sure whether I have the access to do so. Please forward me a guide that how to create the PAT, I'll have a try

**Comment by aazam-gh** [comment]
Sure, here is the guide
https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token

**Comment by bjhuangr** [comment]
I have some concern on provide my personal access token.

Normally, it's not appropriate to use a personal access token in the CI/CD process, instead, use a functional ID's token(with minimum access) to do so.

**Comment by aazam-gh** [comment]
For creating and using the webhook with OBS, we require a personal access token.

    https://openbuildservice.org/help/manuals/obs-user-guide/cha-obs-scm-ci-workflow-integration#sec-obs-obs-scm-ci-workflow-integration-setup-token-authentication-how-to-authenticate-obs-with-scm

**Comment by aazam-gh** [comment]
For the CI/CD process i don't think we require a PAT here since GitHub action uses a $Github token. (Will have to test to confirm)

Update: We cannot avoid using PAT if we want to use the current method. We need both the PATs

**Comment by bjhuangr** [comment]
I'd suggest to create a dedicate github ID for CI/CD purpose, with minimum access to complete the CI/CD tasks. @Bischoff @mfriesenegger @jichenjc @aazam-gh

**Comment by aazam-gh** [comment]
Can you provide any links to creating and setting for the GitHub ID

**Comment by mfriesenegger** [comment]
[GitHub Apps](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/differences-between-github-apps-and-oauth-apps) seem like the preferred way to create tokens used for automation tasks.  I have not read this fully but what I have read quickly seems to be what we should be using.  More research is needed to confirm.

The doc states "only organization owners can manage the settings of GitHub Apps in an organization."  The admins of the OMP are the org owner for feilong.   I do not know who the org owner for mfcloud is.

**Pull request merged**